src/elf/parser.rs

```diff
@@ -296,6 +295,54 @@ impl<'mmap> Cache<'mmap> {
             }
         }
     }
+
+    // Note: This function should really return a reference to
+    //       `self.str2symtab`, but current borrow checker limitations
+    //       effectively prevent us from doing so.
+    fn ensure_str2symtab(&mut self) -> Result<(), Error> {
+        let strtab = self.ensure_strtab()?;
+        let () = self.ensure_symtab()?;
+        // SANITY: The above `ensure_symtab` ensures we have `symtab`
+        //         available.
+        let symtab = self.symtab.as_ref().unwrap();
+
+        match self.str2symtab {
+            Some(_) => Ok(()),
```

Comment:
maybe let's reduce nesting by existing early? 

```
if let Some(_) = self.str2symtab) {
    return Ok();
}

// handle None case
```

Reply:
Sounds good, will change that everywhere.